Autocoders/Python/src/fprime_ac/generators/visitors/SerialHVisitor.py

```diff
@@ -187,18 +186,28 @@ def _get_enum_string_list(self, enum_list):
         enum_list = enum_list[1]
         enum_str_list = []
         for e in enum_list:
-            # No value, No comment
-            if (e[1] is None) and (e[2] is None):
-                s = "%s," % (e[0])
-            # No value, With comment
-            elif (e[1] is None) and (e[2] is not None):
-                s = "{},  // {}".format(e[0], e[2])
-            # With value, No comment
-            elif (e[1] is not None) and (e[2] is None):
-                s = "{} = {},".format(e[0], e[1])
-            # With value and comment
-            elif (e[1] is not None) and (e[2] is not None):
-                s = "%s = %s,  // %s" % (e)
+            # Without value
```

Comment:
There are too many nested statements in my opinion.

Wouldn't something like below be better?

```py
    def _get_enum_string_list(self, enum_list):
        """"""
        enum_tuple = enum_list[0]
        enum_list = enum_list[1]
        enum_str_list = []
        for e in enum_list:
            # No value, No comment
            if (e[1] is None) and (e[2] is None):
                s = f"{e[0]},"
            elif e[1] is None:
                s = f"{e[0]},  // {e[2]}"
            elif e[2] is None:
                s = f"{e[0]} = {e[1]},"
            else:
                s = f"{e} = {e},  // {e}"
            enum_str_list.append(s)

        return (enum_tuple, enum_str_list)
```

Reply:
This would work too, better than the first implementation

Reply:
but it'll take more time to execute, because it's checking some statements that are unnecessary.

Reply:
every time it runs through all the statements even if we only  want the "else" to be  executed. i.e. if it's value and comment then it would run the 3 statements first which increases execution time, whereas my implementation it would ignore the first 3 if-statements and immediately runs the "else".

Its hard to read but easy for the execution

Reply:
Correct me if I'm wrong, but the `else` will never be executed right?

The variable is `None` or it is not, so you can transform the `elif` into `else`.

Once this is done and a final refactoring, you are back to my proposal it seems.

Reply:
`else` is executed if we have either **value** `not None` or **comment** `not None`. 
So basically you're right instead of `elif` we can replace it with `else`

correct

Reply:
I am with @ThibFrgsGmz here.  The nested `if`s make this much harder to review.  I like the existing structure and I'd revert to that.  @ThibFrgsGmz's sample code would also be acceptable.

The other fixes in this PR look good, but the changing of the if structures, although performant, make the code header to understand.